### PR TITLE
[Upgrade Jenkins] Upgrade jenkins docker image

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
This change will build new docker image with `jenkins/jenkins:2.387.1-lts-jdk11` as base. Also install all plugins as part of it 

### Issues Resolved
part of #260 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docker/plugins.txt [hunk 1]
```diff
@@ -23,7 +23,6 @@ generic-webhook-trigger
 ghprb
 git-parameter
 gradle
-greenballs
```
Comment (peterzhuamazon) [inline docker/plugins.txt:26]: Why remove greenballs it is nice!

**Review by peterzhuamazon — APPROVED** [review]
It is great overall on this PR.
Add some comments.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -4,7 +4,6 @@ on:
   push:
     paths:
       - docker/**
-  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (gaiksaya) [reply .github/workflows/docker.yml:7]: Yes! Trying to keep everything to be added via source before someone hits build.

File: docker/plugins.txt (hunk 1 above)
Comment (gaiksaya) [reply docker/plugins.txt:26]: Its not being used with new UI. Also saw below error:
```
WARNING: Illegal reflective access by hudson.plugins.greenballs.GreenBallsPlugin (file:/var/jenkins_home/plugins/greenballs/WEB-INF/lib/greenballs.jar) to field java.awt.Color.value 
WARNING: Please consider reporting this to the maintainers of hudson.plugins.greenballs.GreenBallsPlugin
at hudson.plugins.greenballs.GreenBallFilter.doFilter(GreenBallFilter.java:64)
``` 

There is related issue opened too: https://github.com/jenkinsci/greenballs-plugin/issues/17
image

File: .github/workflows/docker.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply .github/workflows/docker.yml:7]: Thanks.

File: docker/plugins.txt (hunk 1 above)
Comment (peterzhuamazon) [reply docker/plugins.txt:26]: Got it, thanks!

**Pull request merged**